doc/radosgw/vault.rst

```diff
@@ -0,0 +1,79 @@
+===========================
+HashiCorp Vault Integration
+===========================
+
+HashiCorp `Vault`_ can be used as a secure key management service for
+`Server-Side Encryption`_ (SSE-KMS).
+
+#. `Vault authentication`_
+#. `Create a key in Vault`_
+#. `Configure the Ceph Object Gateway`_
+#. `Upload object`_
+
+Vault authentication
+====================
+
+Vault provides several authentication mechanisms. Currently, the Object Gateway
+supports the `token authentication method`_ only.
+
+When authenticating with Vault using the token method, save the token in a
+plain-text file. The path to this file must be provided in the Gateway
+configuration file (see below). For security reasons, ensure the file is
+readable by the Object Gateway only.
+
+Create a key in Vault
+=====================
+
+Generate and save a 256-bit key in Vault. Vault provides several Secret
+Engines, which store, generate, and encrypt data. For instance, create a key
+in the `KV Secrets engine`_ using Vault's command line client::
+
+  export VAULT_ADDR='http://vaultserver:8200'
+  vault kv put secret/myproject/mybucketkey key=$(dd bs=32 count=1 if=/dev/urandom of=/dev/stdout 2>/dev/null | base64)
```

Comment:
It's not clear to me if this needs to be a "v2" versioned secrets engine or not.  Your example and test logic suggest it probably should be.

Reply:
The secret engine we're currently supporting is the "KV Version 2" -- I'll add a comment in the documentation to clarify that. In the next iteration we are planning to support the "transit" engine and we will need to introduce a new config parameter for that (e.g. "rgw crypt vault engine = kv2 | transit...").

Reply:
I just pushed another commit to update the Vault documentation page and clarify only KV version 2 is currently supported.